packages/angular_devkit/build_angular/src/utils/ssg/render.ts

```diff
@@ -0,0 +1,143 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { OutputFile } from 'esbuild';
+import { readFile } from 'node:fs/promises';
+import { extname, posix } from 'node:path';
+import Piscina from 'piscina';
+import type { RenderResult, ServerContext, WorkerData } from './render-worker';
+
+interface SsgOptions {
+  routesFile?: string;
+  discoverRoutes?: boolean;
+  routes?: string[];
+}
+
+interface AppShellOptions {
+  route?: string;
+}
+
+export async function generateSSG(
+  workspaceRoot: string,
+  tsConfigPath: string,
+  appShellOptions: AppShellOptions = {},
+  ssgOptions: SsgOptions = {},
+  outputFiles: Readonly<OutputFile[]>,
+  document: string,
+  inlineCriticalCss?: boolean,
+  maxThreads = 1,
+): Promise<{
+  output: Record<string, string>;
+  warnings: string[];
+  errors: string[];
+}> {
+  const allRoutes = await getAllRoutes(tsConfigPath, appShellOptions, ssgOptions);
+  const outputFilesForWorker: Record<string, string> = {};
+
+  for (const { text, path } of outputFiles) {
+    switch (extname(path)) {
+      case '.mjs': // Contains the server runnable application code.
+      case '.css': // Global styles for critical CSS inlining.
+        outputFilesForWorker[path] = text;
+        break;
+    }
+  }
+
+  const renderWorker = new Piscina({
+    filename: require.resolve('./render-worker'),
+    maxThreads: allRoutes.size > maxThreads ? maxThreads : 1,
```

Comment:
```suggestion
    maxThreads: Math.min(allRoutes.size, maxThreads),
```
Or is it intended to only use one thread if less than maxThreads?

Reply:
The intention was indeed to use a single worker if it’s less than maxThreads.